src/commonMain/kotlin/exchange.dydx.abacus/processor/router/skip/SkipProcessor.kt

```diff
@@ -27,6 +27,13 @@ internal class SkipProcessor(
 
     var skipTokens: Map<String, Map<String, List<Map<String, Any>>>>? = null
     override var exchangeDestinationChainId: String? = null
+    override var selectedChainType: ChainType? = ChainType.EVM
+        set(value) {
+            if (field != value) {
+                field = value
+                internalState.chains = chainOptions()
```

Comment:
will this force the abacus state to update? since we're not triggering a a state update for updating the `selectedChainType`

we might want to make this a piece of state so the recalculation happens automatically

Reply:
It does seem like Rui recently updated so that we use `internalState.chains` directly for the state output:
https://github.com/dydxprotocol/v4-abacus/blob/main/src/commonMain/kotlin/exchange.dydx.abacus/output/input/TransferInput.kt#L44